### ISSUE TITLE
Investigate crash

### ISSUE THREAD

**Issue by recmo** [worldcoin/signup-sequencer#279]
## Error
```
Sequencer potentially deadlocked, terminating.
```

## Metadata
|key|value|
|--|--|
|**version**|signup-sequencer 0.1.0
a2dc5980b1ccc163e4bfc71c094353bd09847dcd 2022-11-11
x86_64-unknown-linux-musl 2022-11-11
Remco Bloemen <remco@worldcoin.org>:Lucas Ege <lucas@worldcoin.org>
https://github.com/worldcoin/signup-sequencer
Template for a Rust CLI tool|
|**location**|/src/src/app.rs:169:13|

## SpanTrace

<details>

```
SpanTrace:
   0: signup_sequencer::app::insert_identity
             at src/app.rs:151
   1: signup_sequencer::server::api_request
           with request=Request { method: POST, uri: /insertIdentity, version: HTTP/1.1, headers: {"host": "signup.crypto.worldcoin.dev", "user-agent": "axios/0.27.2", "content-length": "73", "accept": "application/json, text/plain, */*", "authorization": "Basic emVyMHByMDBmMjAyMjo1S0VtQjZtUFpHZnlSM3QyRnhkQg==", "content-type": "application/json", "x-datadog-parent-id": "7998491055645856469", "x-datadog-sampling-priority": "1", "x-datadog-trace-id": "7254124415775490356", "x-forwarded-for": "10.0.11.245", "x-forwarded-host": "signup.crypto.worldcoin.dev", "x-forwarded-port": "80", "x-forwarded-proto": "http", "x-forwarded-server": "traefik-5bfc9f96d-b4rtr", "x-real-ip": "10.0.11.245", "accept-encoding": "gzip"}, body: Body(Streaming) } http.uri=/insertIdentity http.method=POST
             at src/server.rs:159
```
</details>

**Comment by recmo** [comment]
Log entry: https://app.datadoghq.com/logs?query=env%3Aprod%20service%3Asignup-sequencer&cols=host%2Cservice&event=AQAAAYR06RGtkmWa-gAAAABBWVIwNlIxbUFBQnJXaXpSX285enVRQVc&index=%2A&messageDisplay=inline&saved-view-id=1320295&stream_sort=time%2Cdesc&viz=stream&from_ts=1668408706753&to_ts=1668408743581&live=false

**Comment by recmo** [comment]
This is a timeout on the tree lock in `insert_identity`. Known problem when load exceeds capacity, should be fixed by #258
